File: go/867. Transpose Matrix.go
// runtime 32ms
func transpose(A [][]int) [][]int {
    n := len(A)
    m := len(A[0])
    T := make([][]int, m)

    for i := 0; i < m; i++ {
        T[i] = make([]int, n)
    }

    for i, Ai := range A {
        for j, v := range Ai {
            T[j][i] = v
        }
    }

    return T
}

File: java/867 Transpose Matrix.java
// Runtime 2ms
class Solution {
    public int[][] transpose(int[][] A) {
        int [][] tran = new int [A[0].length][A.length];
        for(int i=0;i<A.length;i++){
            for(int j=0;j<A[0].length;j++){
                tran[j][i] = A[i][j];
            }
        }
        return tran;
    }

}